modules/diskq/qdisk.c

```diff
@@ -63,15 +60,9 @@ typedef union _QDiskFileHeader
     gint64 write_head;
     gint64 length;
 
-    gint64 qout_ofs;
-    gint32 qout_len;
-    gint32 qout_count;
-    gint64 qbacklog_ofs;
-    gint32 qbacklog_len;
-    gint32 qbacklog_count;
-    gint64 qoverflow_ofs;
-    gint32 qoverflow_len;
-    gint32 qoverflow_count;
+    QDiskQueuePosition qout_pos;
+    QDiskQueuePosition qbacklog_pos;
+    QDiskQueuePosition qoverflow_pos;
```

Comment:
I've validated that the offset of every field remained the same on my machine, but I'm not sure about other platforms. I find this refactor step a bit risky.

As we pointed out IRL, the manual padding in`QDiskFileHeader` is incorrect (it was incorrect in the original version as well), an addition 1-byte padding is missing after `guint8 _pad1;`

Reply:
@MrAnno: we can do further tests on various cpu architectures... ( as each 'group' is 128=64+32+32 bytes, I don't think it would be an issue, more details:
 * in C, there are no lead padding for structs
 * alignment will be 64 (as the widest type is uint64_t in the struct)
 * as the other two members are 32 bits, I think there is no chance for having a padding inside the struct (nor trailing padding)
 * we don't have to take structure reordering into account (but it wouldn't cause any issue)
 )